librz/bin/p/bin_java.c

```diff
@@ -1,227 +1,259 @@
+	rz_list_join(list, tmp);
+	rz_list_free(tmp);
+
+	tmp = rz_bin_java_class_const_pool_as_symbols(jclass);
+	rz_list_join(list, tmp);
+	rz_list_free(tmp);
+	return list;
 }
 
-static bool check_buffer(RzBuffer *b) {
-	if (rz_buf_size(b) > 32) {
-		ut8 buf[4];
-		rz_buf_read_at(b, 0, buf, sizeof(buf));
-		if (!memcmp(buf, "\xca\xfe\xba\xbe", 4)) {
-			int off = rz_buf_read_be32_at(b, 4 * sizeof(int));
-			int version = rz_buf_read_be16_at(b, 6);
-			if (off > 0 && version < 1024) {
-				return true;
-			}
-		}
+static RzList *fields(RzBinFile *bf) {
+	RzBinJavaClass *jclass = rz_bin_file_get_java_class(bf);
+	if (!jclass) {
+		return NULL;
 	}
-	return false;
+
+	return rz_bin_java_class_fields_as_binfields(jclass);
 }
 
-static int retdemangle(const char *str) {
-	return RZ_BIN_NM_JAVA;
+static RzList *libs(RzBinFile *bf) {
+	RzBinJavaClass *jclass = rz_bin_file_get_java_class(bf);
+	if (!jclass) {
+		return NULL;
+	}
+
+	return rz_bin_java_class_as_libraries(jclass);
 }
 
 static RzBinAddr *binsym(RzBinFile *bf, int sym) {
-	return rz_bin_java_get_entrypoint(bf->o->bin_obj, sym);
-}
+	RzBinJavaClass *jclass = rz_bin_file_get_java_class(bf);
+	if (!jclass) {
+		return NULL;
+	}
 
-static RZ_BORROW RzList *lines(RzBinFile *bf) {
-	return NULL;
-#if 0
-	char *file = bf->file? strdup (bf->file): strdup ("");
-	RzList *list = rz_list_newf (free);
-	// XXX the owner of this list should be the plugin, so we are leaking here
-	file = rz_str_replace (file, ".class", ".java", 0);
-	/*
-	   int i;
-	   RzBinJavaObj *b = bf->o->bin_obj;
-	   for (i=0; i<b->lines.count; i++) {
-	        RzBinDwarfRow *row = RZ_NEW0(RzBinDwarfRow);
-	        rz_bin_dwarf_line_new (row, b->lines.addr[i], file, b->lines.line[i]);
-	        rz_list_append (list, row);
-	   }*/
-	free (file);
-	return list;
-#endif
+	return rz_bin_java_class_resolve_symbol(jclass, sym);
 }
 
-static RzList *sections(RzBinFile *bf) {
-	return rz_bin_java_get_sections(bf->o->bin_obj);
+static RzList *entrypoints(RzBinFile *bf) {
+	RzBinJavaClass *jclass = rz_bin_file_get_java_class(bf);
+	if (!jclass) {
+		return NULL;
+	}
+
+	return rz_bin_java_class_entrypoints(jclass);
 }
 
-static RzList *imports(RzBinFile *bf) {
-	return rz_bin_java_get_imports(bf->o->bin_obj);
+static RzList *strings(RzBinFile *bf) {
+	RzBinJavaClass *jclass = rz_bin_file_get_java_class(bf);
+	if (!jclass) {
+		return NULL;
+	}
+
+	return rz_bin_java_class_strings(jclass);
 }
 
-static RzList *fields(RzBinFile *bf) {
-	return NULL; // rz_bin_java_get_fields (bf->o->bin_obj);
+static int demangle_type(const char *str) {
+	return RZ_BIN_NM_JAVA;
 }
 
-static RzList *libs(RzBinFile *bf) {
-	return rz_bin_java_get_lib_names(bf->o->bin_obj);
+static char *enrich_asm(RzBinFile *bf, const char *asm_str, int asm_len) {
+	RzBinJavaClass *jclass = rz_bin_file_get_java_class(bf);
+	if (!jclass) {
+		return NULL;
+	}
+	for (int i = 0; i < asm_len; ++i) {
+		if (asm_str[i] == 'c' && !strncmp(asm_str + i, "constant_pool.", 14)) {
```

Comment:
let the compiler be smart for you
```suggestion
		if (!strncmp(asm_str + i, "constant_pool.", 14)) {
```

Reply:
Also, put `"constant_pool."` in a var/define and use `strlen(var)` instead of `14`. The compiler will optimize away  the strlen anyway and I don't have to check that `constant_pool.` is really of 14 chars :D